SPHINXsys/src/shared/common/small_vectors.cpp

```diff
@@ -0,0 +1,217 @@
+		minv(1, 2) = (A(1, 0) * A(0, 2) - A(0, 0) * A(1, 2)) * invdet;
+		minv(2, 0) = (A(1, 0) * A(2, 1) - A(2, 0) * A(1, 1)) * invdet;
+		minv(2, 1) = (A(2, 0) * A(0, 1) - A(0, 0) * A(2, 1)) * invdet;
+		minv(2, 2) = (A(0, 0) * A(1, 1) - A(1, 0) * A(0, 1)) * invdet;
+
+		return minv;
+	}
+	//=================================================================================================//
+	Mat2d getAverageValue(const Mat2d& A, const Mat2d& B)
+	{
+		Mat2d C(1.0);
+		for (int i = 0; i < 2; i++)
+		{
+			for (int j = 0; j < 2; j++)
+			{
+				C(i, j) = 2.0 * A(i, j) * B(i, j) / (A(i, j) + B(i, j) + TinyReal);
+			}
+		}
+		return C;
+	}
+	//=================================================================================================//
+	Mat3d getAverageValue(const Mat3d& A, const Mat3d& B)
+	{
+		Mat3d C(1.0);
+		for (int i = 0; i < 3; i++)
+		{
+			for (int j = 0; j < 3; j++)
+			{
+				C(i, j) = 2.0 * A(i, j) * B(i, j) / (A(i, j) + B(i, j) + TinyReal);
+			}
+		}
+		return C;
+	}
+	//=================================================================================================//
+	Mat2d inverseCholeskyDecomposition(const Mat2d& A)
+	{
+		Mat2d lower(0);
+		int n = 2;
+		/** Decomposing a matrix into Lower Triangular. */
+		for (int i = 0; i < n; i++)
+		{
+			for (int j = 0; j < (i + 1); j++)
+			{
+				double sum = 0;
+				for (int k = 0; k < j; k++)
+				{
+					sum += lower(i, k) * lower(j, k);
+				}
+				if (i == j)
+				{
+					lower(i, j) = sqrt(A(i, i) - sum);
+				}
+				else
+				{
+					lower(i, j) = (1.0 / lower(j, j) * (A(i, j) - sum));
+				}
+			}
+		}
+		Mat2d inverse_lower = getInverse(lower);
+		return inverse_lower;
+	}
+	//=================================================================================================//
+	Mat3d inverseCholeskyDecomposition(const Mat3d& A)
+	{
+		Mat3d lower(0);
+		int n = 3;
+		/** Decomposing a matrix into Lower Triangular. */
+		for (int i = 0; i < n; i++)
+		{
+			for (int j = 0; j < (i + 1); j++)
+			{
+				double sum = 0;
+				for (int k = 0; k < j; k++)
+				{
+					sum += lower(i, k) * lower(j, k);
+				}
+				if (i == j)
+				{
+					lower(i, j) = sqrt(A(i, i) - sum);
+				}
+				else
+				{
+					lower(i, j) = (1.0 / lower(j, j) * (A(i, j) - sum));
+				}
+			}
+		}
+		Mat3d inverse_lower = getInverse(lower);
+		return inverse_lower;
+	}
+	//=================================================================================================//
+	Mat2d getTransformationMatrix(const Vec2d& direction_of_y)
+	{
+		Mat2d transformation_matrix(0.0);
+		transformation_matrix[0][0] = direction_of_y[1];
+		transformation_matrix[0][1] = -direction_of_y[0];
+		transformation_matrix[1][0] = direction_of_y[0];
+		transformation_matrix[1][1] = direction_of_y[1];
+		return transformation_matrix;
+	}
+	//=================================================================================================//
+	Mat3d getTransformationMatrix(const Vec3d& direction_of_z)
```

Comment:
suggest to have unit test
The code is likely to be error prone

Reply:
OK. The unit test should be added in future.